mountpoint-s3/src/inode.rs

```diff
@@ -189,6 +189,39 @@ impl Superblock {
         self.inner.lookup(client, inode.parent(), inode.name().as_ref()).await
     }
 
+    /// Set the attributes for an inode
+    pub async fn setattr<OC: ObjectClient>(
+        &self,
+        _client: &OC,
+        ino: InodeNo,
+        atime: Option<OffsetDateTime>,
+        mtime: Option<OffsetDateTime>,
+    ) -> Result<LookedUp, InodeError> {
+        let inode = self.inner.get(ino)?;
+        let mut sync = inode.get_mut_inode_state()?;
+
+        if sync.write_status == WriteStatus::Remote {
+            return Err(InodeError::SetAttrNotPermittedOnRemoteInode(ino));
+        }
+
+        // Should be impossible since local file stat never expire.
+        if !sync.stat.is_valid() {
+            warn!(?ino, "local inode stat already expired");
+            return Err(InodeError::SetAttrOnExpiredStat(ino));
+        }
```

Comment:
Not really sure how we should handle this case, another option is doing lookup to update the validity, but it should not happen so I think it makes sense to return an error immediately.